doc/arch/adr-004-geoprocessing-api.md

```diff
@@ -0,0 +1,56 @@
+##### 3. Not result in duplicated code
+The web platform needs to have access to unauthenticated endpoints.  This could theoretically be accomplished through domain white listing within the DRF endpoints or the existing endpoints.  Regardless, the implementation should be generalized in one specific place that the API and platform utilize.
+##### 4. Enforce size and shape validation
+The front end code which submits analysis jobs is responsible for the majority of shape validation.  This will need to be included in the API, as well as for additional potential conflicts like coordinate systems and being contained within the CONUS.
+
+#### Options
+##### Reuse existing endpoints
+Our existing endpoints `/start/analyze/*` and `/start/rwd` have the exact functionality that is needed, but are limited by specific implementation details:
+* Do not require authentication
+* Validation is largely handled by front end submission routine
+* Paths and naming were not selected in the context of a public API
+* CORS is not enabled
+
+##### Create new DRF based API endpoints
+DRF is already used in the application and has support for authentication systems such as [Token Auth](http://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication), as well as some helper views for manging creation tokens.  It also supports a pluggable auth system that includes broader 3rd party set of auth providers, including more potentially sophisticated techniques such as JWT.  It does not have any default usage logging, so this would need to be handled manually, but it does support a mechanism for [throttling](http://www.django-rest-framework.org/api-guide/throttling/), which can be customized.  Azavea's [Climate Change API](https://github.com/azavea/climate-change-api) uses this library.
+
+
+##### Use AWS API Gateway to proxy API requests
+AGW could pass requests to existing or new endpoints and comes with a very robust system feature set including usage plans, rate limiting, logging and versioning.  The tooling and execution around this service is relatively unknown on the team.  Also, it would need to authorize requests based on authentication to the application, requiring custom code.
+
+
+### Decision
+We will use the existing Django REST Framework to implement new API endpoints, secured by the default Token Authentication provided by the framework.  We will throttle requests to a configurable amount per minute.  We will have to add an account page that lists the users API key and allows it to be invalided/regenerated.  Validation will be encoded into the endpoints and the application/environment will have a user associated with it, and will use that user's API key against the same endpoints. Basic logging of requests will be done within the Django model/database with attributes confirmed by the client.
```

Comment:
Looking around, there seems to be a [well-used plugin](http://drf-tracking.readthedocs.io/en/latest/) for tracking requests in the database. Could be an easy way to add that feature.